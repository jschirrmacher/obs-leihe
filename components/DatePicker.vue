<script setup lang="ts">
import VueDatePicker from "@vuepic/vue-datepicker"
import "@vuepic/vue-datepicker/dist/main.css"

const date = defineModel<Date>()
const locale = navigator.language
</script>

<template>
  <VueDatePicker
    v-model="date"
    :enable-time-picker="false"
    :locale="locale"
    :format="(date: Date) => date.toLocaleDateString()"
    select-text="Ok"
    cancel-text="Abbrechen"
    :teleport="true"
    hide-input-icon
    class="date-selection"
  />
</template>

<style>
.date-selection {
  width: 9em;
}
</style>
