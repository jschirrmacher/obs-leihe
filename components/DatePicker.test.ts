import { describe, expect, it } from "vitest"
import { mountSuspended } from "@nuxt/test-utils/runtime"
import DatePicker from "./DatePicker.vue"

describe("DatePicker Component", () => {
  it("should render as expected", async () => {
    const component = await mountSuspended(DatePicker, {})
    expect(component.html()).toMatchSnapshot()
  })
})
