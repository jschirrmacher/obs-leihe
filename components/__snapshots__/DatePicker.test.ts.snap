// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DatePicker Component > should render as expected 1`] = `
"<div class="dp__main dp__theme_light date-selection" data-datepicker-instance="">
  <div>
    <!---->
    <div class="dp__input_wrap">
      <!----><input data-test="dp-input" class="dp__pointer dp__input_readonly dp__input dp__input_reg" inputmode="none" placeholder="" autocomplete="off" aria-label="Datepicker input">
      <div>
        <!---->
        <!---->
      </div>
      <!---->
      <!---->
    </div>
  </div>
  <!--teleport start-->
  <!--teleport end-->
</div>"
`;
